docs/resources/upgrading.rst

```diff
@@ -0,0 +1,173 @@
+    - ``@edgedb/generate``
+    - ``@gel/generate``
+
+Check your project task runners and update them accordingly.
+
+Upgrading instances
+===================
+
+To take advantage of the new features in Gel v6, you'll need to upgrade your instances to the latest version.
+
+Cloud instances
+---------------
+
+If you're using a hosted instance on Gel Cloud, you can upgrade your instance by clicking on the "Upgrade" button in the Gel Cloud console, or with the CLI.
+
+.. code-block:: bash
+
+  $ gel instance upgrade <my-org/my-instance-name> --to-latest
+
+Local instances
+---------------
+
+If you have local instances that you've intialized with the CLI using ``edgedb project init``, you can upgrade them easily with the CLI.
+
+.. code-block:: bash
+
+  gel project upgrade --to-latest
+
+This will upgrade the project instance to the latest version of Gel and also update the |gel.toml| server-version value to the latest version.
+
+Remote instances
+----------------
+
+To upgrade a remote instance, we recommend the following dump-and-restore process:
+
+1. Gel v6.0 supports PostgreSQL 14 or above. Verify your PostgreSQL version before upgrading Gel. If you're using Postgres 13 or below, upgrade Postgres first.
+
+2. Spin up an empty 6.0 instance. You can use one of our :ref:`deployment guides <ref_guide_deployment>`.
+
+   For Debian/Ubuntu, when adding the Gel package repository, use this command:
+
+   .. code-block:: bash
+
+       $ echo deb [signed-by=/usr/local/share/keyrings/gel-keyring.gpg] \
+           https://packages.geldata.com/apt \
+           $(grep "VERSION_CODENAME=" /etc/os-release | cut -d= -f2) main \
+           | sudo tee /etc/apt/sources.list.d/gel.list
+       $ sudo apt-get update && sudo apt-get install gel-6
+
+   For CentOS/RHEL, use this installation command:
+
+   .. code-block:: bash
+
+       $ sudo yum install gel-6
+
+   In any required ``systemctl`` commands, replace ``edgedb-server-5`` with ``gel-server-6``.
+
+   For Docker setups, use the ``6`` or other appropriate tag.
+
+   .. note::
+
+     The new instance will have a different DSN, including a different port number. Take note of the full DSN of the new instance as you'll need it to restore your database, and update your application to use the new DSN in further steps.
+
+3. Take your application offline, then dump your v5.x database with the CLI:
+
+   .. code-block:: bash
+
+       $ gel dump --dsn <old dsn> --all --format dir my_database.dump/
+
+   This will dump the schema and contents of your current database to a directory on your local disk called ``my_database.dump``. The directory name isn't important.
+
+4. Restore to the new, empty v6 instance from the dump:
+
+   .. code-block:: bash
+
+       $ gel restore --all my_database.dump/ --dsn <new dsn>
+
+   Once the restore is complete, update your application to connect to the new instance.
+
+   This process will involve some downtime, specifically during steps 2 and 3.
+
+GitHub Action
```

Comment:
I think we're still waiting on https://github.com/geldata/setup-gel/pull/39 before we can actually suggest this.